### PR TITLE
fix linter warnings and some headers

### PR DESCRIPTION
**Pull request by conao3**
Hi! I found many linter warnings so I try to solve them.
But many warnings remained.  I fix easy warnings only.

## Warnings

``` emacs-lisp
Lisp symbol ‘regexp-opt’ should appear in quotes (emacs-lisp-checkdoc)
Lisp symbol ‘regexp-quote’ should appear in quotes (emacs-lisp-checkdoc)
Lisp symbol ‘text-mode’ should appear in quotes (emacs-lisp-checkdoc)
Name lisp should appear capitalized as Lisp (emacs-lisp-checkdoc)
```

## Headers

Some headers wrote Created date as <year>-<mon>-<day> format. Follow smartparens.el.

Fix header comment to follow smartparens.el

Remove unneeded version/keyword/url comment header (should refer main smartparens.el).

### CODE REVIEW

File: smartparens.el [hunk 1]
```diff
@@ -9582,16 +9584,14 @@ has a pair definition.  If so, we insert the closing pair."
...
+  '(sp-wrap-point
     sp-wrap-mark
     sp-last-wrapped-region
     sp-pair-overlay-list
     sp-wrap-overlays
     sp-wrap-tag-overlays
     sp-last-operation
-    sp-previous-point
-    )
+    sp-previous-point)
```
Comment (Fuco1) [inline smartparens.el]: The paren here is intentionally on next line so diffs are nicer: if one thing is added the diff is "+1" instead of "-1/+2".

Comment (conao3) [reply smartparens.el]: Revert this change.

**Comment by conao3** [comment]
I reverted your mentioned diff. There are any other concerns?

**Comment by Fuco1** [comment]
I'll need to do another review, it's a huuuge change set.  Thanks for your patience!

**Comment by conao3** [comment]
Thanks reply! It's just a ping. I thank you very much for maintaining many packages.

**Comment by conao3** [comment]
Any concerns?

**Comment by Hi-Angel** [comment]
The list of warnings that is in the PR description should go instead to the commit body. Because once it's merged, it's highly unlikely anybody's good read the PR, but people do read git-log.

----------------

With that said, @Fuco1 ping? I also noted that smartparens produces various warnings with native compilation, I think this PR would be a good start *(even though I presume it doesn't fix byte-compilation warnings)*

**Comment by Fuco1** [comment]
@Hi-Angel The fixes here are 90% from check doc so it only changes documentation strings and not actual code.  It is still worth going over though.

**Comment by conao3** [comment]
Hi, sorry for this outdated PR. I am in a busy these days and can't spend time on this PR, so I will close it for now.
I will declare CC0 for commit in this PR, even though it may not be necessary.

**Comment by Fuco1** [comment]
Don't worry, I am sorry for not looking into it sooner.  It's still valuable and I will definitely go through it (I'm now slowly going over all pull requests over all my projects)

**Pull request closed without merging**